### PR TITLE
Main popup

### PR DESCRIPTION
**Pull request by Dauntless-Jacobson**

### CODE REVIEW

**Comment by Dauntless-Jacobson** [comment]
mainpopargs

**Comment by Dauntless-Jacobson** [comment]
i will utilize the it buttons which is why it look like that currently there is a pr to fix them

File: src/web-components/MainPopup/index.ts [hunk 1]
```diff
@@ -0,0 +1,157 @@
...
+    }
+
+    .popup-actions button:nth-child(2) {
+        background-color: #6c757d;
+    }
+
+    .popup-actions button:nth-child(2):hover {
+        background-color: #5a6268;
+    }
+    `;
```
Comment (bigirishlion) [inline src/web-components/MainPopup/index.ts:103]: Let's continue the format of having the CSS in a separate file

File: src/web-components/MainPopup/index.ts [hunk 2]
```diff
@@ -0,0 +1,157 @@
...
+ *     open="true" 
+ *     message="Are you sure?" 
+ *     secondMessage="This action cannot be undone."
+ *     buttonConfirm="Confirm" 
+ *     buttonCancel="Cancel" 
+ *     .confirmCallback=${() => console.log('Confirmed!')}>
+ * </main-popup>
+ * ```
+ */
+@customElement('main-popup')
```
Comment (bigirishlion) [inline src/web-components/MainPopup/index.ts:26]: Let's rename this web component to "confirm-popup"

File: src/web-components/MainPopup/index.ts [hunk 3]
```diff
@@ -0,0 +1,157 @@
...
+    @property({ type: Boolean }) open = false;
+    @property({ type: String }) message = '';
+    @property({ type: String }) secondMessage = '';
+    @property({ type: String }) buttonConfirm = '';
+    @property({ type: String }) buttonCancel = '';
+    @property({ type: Function }) confirmCallback: (() => void) | null = null;
+
+    closePopup() {
+        this.open = false;
+    }
```
Comment (bigirishlion) [inline src/web-components/MainPopup/index.ts:113]: Let's also add in an optional function for `handleCancel` that is called so we can allow the dev to hook into the event

File: src/stories/MainPopup.stories.ts [hunk 4]
```diff
@@ -0,0 +1,51 @@
+import { html, TemplateResult } from 'lit';
+import '../web-components/MainPopup/index.js';
+
+export default {
+    title: 'Components/Main Popup',
+    component: 'main-popup',
```
Comment (bigirishlion) [inline src/stories/MainPopup.stories.ts:6]: Let's change to Confirm Popup (confirm-popup)

**Review by bigirishlion — CHANGES_REQUESTED** [review]
🪣

File: src/web-components/ConfirmPopup/index.ts [hunk 5]
```diff
@@ -0,0 +1,91 @@
...
+export class ConfirmPopup extends LitElement {
+    static styles = ConfirmPopupStyles;
+    @property({ type: Boolean }) open = false;
+    @property({ type: String }) message = '';
+    @property({ type: String }) secondMessage = '';
+    @property({ type: String }) buttonConfirm = '';
+    @property({ type: String }) buttonCancel = '';
+    @property({ type: Function }) confirmCallback: (() => void) | null = null;
+    @property({ type: Function }) confirmCancel: (() => void) | null = null;
+
```
Comment (bigirishlion) [inline src/web-components/ConfirmPopup/index.ts:37]: Extra line

File: src/web-components/ConfirmPopup/index.ts [hunk 6]
```diff
@@ -0,0 +1,91 @@
...
+ * - **confirmCallback**: (Function|null) A callback function to be called when the confirm button is clicked.
+ * 
+ * ### Example:
+ * ```html
+ * <main-popup 
+ *     open="true" 
+ *     message="Are you sure?" 
+ *     secondMessage="This action cannot be undone."
+ *     buttonConfirm="Confirm" 
+ *     buttonCancel="Cancel"
```
Comment (bigirishlion) [inline src/web-components/ConfirmPopup/index.ts:22]: I think all of these have period `.` in front correct?

File: src/web-components/ConfirmPopup/index.ts [hunk 7]
```diff
@@ -0,0 +1,91 @@
...
+        this.open = true;
+    }
+    private handleConfirm() {
+        if (this.confirmCallback) {
+            this.confirmCallback();
+        }
+        this.closePopup();
+    }
+    private handleCancel() {
+        if (this.confirmCallback) {
```
Comment (bigirishlion) [inline src/web-components/ConfirmPopup/index.ts:58]: should this be `if (this.confirmCancel)`?

File: src/web-components/ConfirmPopup/index.ts [hunk 8]
```diff
@@ -0,0 +1,91 @@
...
+@customElement('confirm-popup')
+export class ConfirmPopup extends LitElement {
+    static styles = ConfirmPopupStyles;
+    @property({ type: Boolean }) open = false;
+    @property({ type: String }) message = '';
+    @property({ type: String }) secondMessage = '';
+    @property({ type: String }) buttonConfirm = '';
+    @property({ type: String }) buttonCancel = '';
+    @property({ type: Function }) confirmCallback: (() => void) | null = null;
+    @property({ type: Function }) confirmCancel: (() => void) | null = null;
```
Comment (bigirishlion) [inline src/web-components/ConfirmPopup/index.ts:36]: `confirmCancel` is confusing. Maybe we call this `onCancel` and change `confirmCallback` to `onConfirm`

**Review by bigirishlion — CHANGES_REQUESTED** [review]
👍

File: src/stories/ConfirmPopup.stories.ts [hunk 9]
```diff
@@ -0,0 +1,55 @@
...
+
+export default {
+    title: 'Components/Confirm Popup',
+    component: 'confirm-popup',
+    argTypes: {
+        open: { control: 'boolean', description: 'Controls whether the popup is open or closed' },
+        message: { control: 'text', description: 'Main message to display in the popup' },
+        secondMessage: { control: 'text', description: 'Secondary message to display in the popup' },
+        buttonConfirm: { control: 'text', description: 'Text for the confirm button' },
+        buttonCancel: { control: 'text', description: 'Text for the cancel button' },
```
Comment (bigirishlion) [inline src/stories/ConfirmPopup.stories.ts:12]: Can we change these to `confirmButtonLabel` and `cancelButtonLabel`?

File: src/web-components/ConfirmPopup/index.ts [hunk 10]
```diff
@@ -0,0 +1,85 @@
+import { LitElement, html, css } from 'lit';
+import { customElement, property } from 'lit/decorators.js';
+import { ConfirmPopupStyles } from './styles';
+/**
+ * `MainPopup` is a custom popup component for displaying messages and handling user actions.
```
Comment (bigirishlion) [inline src/web-components/ConfirmPopup/index.ts:5]: `ConfirmPopup`

File: src/web-components/ConfirmPopup/index.ts [hunk 11]
```diff
@@ -0,0 +1,85 @@
...
+ * 
+ * ### Properties:
+ * - **open**: (boolean) Indicates whether the popup is open or closed.
+ * - **message**: (string) The primary message to display in the popup.
+ * - **secondMessage**: (string) The secondary message to display in the popup.
+ * - **buttonConfirm**: (string) The label for the confirm button.
+ * - **buttonCancel**: (string) The label for the cancel button.
```
Comment (bigirishlion) [inline src/web-components/ConfirmPopup/index.ts:12]: Change if changed above

File: src/web-components/ConfirmPopup/index.ts [hunk 12]
```diff
@@ -0,0 +1,85 @@
...
+ * - **onConfirm**: (Function|null) A callback function to be called when the confirm button is clicked.
+ * - **onCancel**: (Function|null) A callback function to be called when the cancel button is clicked.
+ * 
+ * ### Example:
+ * ```html
+ * <main-popup
```
Comment (bigirishlion) [inline src/web-components/ConfirmPopup/index.ts:18]: `<confirm-popup`

**Review by bigirishlion — CHANGES_REQUESTED** [review]
📦

**Review by bigirishlion — APPROVED** [review]
🍨

**Pull request merged**